### DISCUSSION TITLE
Newly spawned sprites briefly appear at default Transform

### DISCUSSION THREAD

**Discussion by Jazarro** [bevyengine/bevy#5316] [Q&A] [answered]
Hey, I'm making a 2D game with sprites. Whenever I spawn a new sprite, it briefly (for 1 frame, presumably) appears at the wrong Transform, I assume at the origin at default scaling. This means that whenever I spawn a new sprite, it briefly flashes in the middle of my screen, before moving into its proper intended place.

Does anyone know what causes this and how to avoid it? My best guess right now is it has something to do with the globalTransform being updated one frame later or something.

**Comment by Nilirad** [comment, marked as answer]
It looks like you are spawning a sprite first and then changing its `Transform` component in a later stage.
If you set the transform right when you are setting up the `SpriteBundle` it shouldn't happen.

**Reply by Jazarro** [reply]
I'm setting the transform right away. Here's an excerpt:

```Rust
let mut transform = Transform::default();
transform.translation.x = pos.x as f32 + dimens.x as f32 * 0.5;
transform.translation.y = pos.y as f32 + dimens.y as f32 * 0.5;
transform.translation.z = depth.z();
transform.scale = Vec3::new(1. / 128., 1. / 128., 1.0);
commands.spawn_bundle(SpriteSheetBundle {
    texture_atlas,
    transform,
    sprite: TextureAtlasSprite { index: 0, ..default() },
    ..default()
})
```

The incorrect scale is particularly noticeable. At the default scale, my pictures fill up the entire screen.

**Reply by Jazarro** [reply]
I tried setting an initial `GlobalTransform`, and that did work. I got the impression that the global transform was maintained internally and I should not write to it manually, but perhaps in this case I was wrong?

**Reply by DJMcNab** [reply]
The most likely issue here is that you are spawning your sprite between hierarchy propagation and extract. I'm not sure of the exact order.

What stage is your spawning system in.

Can you try to spawn your sprite in `Update`?

**Reply by Jazarro** [reply]
That solved it! I was using custom stages, using iyes_loopless, which apparently ran all the way at the end of the schedule. 

After changing
`.add_stage("set_intent", SystemStage::parallel())`
to
`.add_stage_after(Update,"set_intent", SystemStage::parallel())`
the incorrect behaviour stopped. Thank you!
